File: game/src/main/java/io/github/fernthedev/secondgame/main/inputs/keyboard/KeyButton.java
package io.github.fernthedev.secondgame.main.inputs.keyboard;

import lombok.*;

@RequiredArgsConstructor
@ToString
public class KeyButton {

    @Getter
    private final int charCode;

    @Getter
    @Setter
    @NonNull
    private boolean held;

    private int pressed = 0;

    public boolean isPressed() {
        if (held && pressed != 0) {
            pressed--;
            return true;
        }
        return false;
    }

}

File: game/src/main/java/io/github/fernthedev/secondgame/main/inputs/keyboard/KeyButton.kt
package io.github.fernthedev.secondgame.main.inputs.keyboard


class KeyButton(
    val charCode: Int = 0,
    var held: Boolean = false
) {

    private var pressed = 0

    fun isPressed(): Boolean {
        if (held && pressed != 0) {
            pressed--
            return true
        }
        return false
    }

}